proc-macros/src/render_client.rs

```diff
@@ -68,6 +68,36 @@ impl RpcDescription {
 		Ok(trait_impl)
 	}
 
+	fn patch_result_error(&self, ty: &syn::Type) -> syn::Type {
+		let mut path = match ty {
+			syn::Type::Path(path) => path.clone(),
+			_ => panic!("Client only supports bare or Result values: {:?}", ty),
+		};
+
+		let Some(first_segment) = path.path.segments.first_mut() else {
+			return syn::Type::Path(path);
+		};
+
+		if first_segment.ident != "Result" {
+			return syn::Type::Path(path);
+		}
+
+		let args = match first_segment.arguments {
+			PathArguments::AngleBracketed(AngleBracketedGenericArguments { ref mut args, .. }) => args,
+			_ => unreachable!("Unexpected Result structure"),
+		};
+
+		let error = args.last_mut().unwrap();
```

Comment:
nit: Should we check here that we only have 2 arguments? Tho, I'm wondering how this would play into some syntax sugar like `type Result<T> = std::Result<T, MyErr>`?